provisioning/service/src/Config/IndividualEnrollment.cs

```diff
@@ -250,17 +254,25 @@ public string DeviceId
         /// Attestation Mechanism
         /// </summary>
         [JsonProperty(PropertyName = "attestation")]
-        private AttestationMechanism _Attestation;
+        private AttestationMechanism _attestation;
         [JsonIgnore]
         public Attestation Attestation
         {
             get
             {
-                return _Attestation.GetAttestation();
+                return _attestation.GetAttestation();
             }
             set
             {
-                _Attestation = new AttestationMechanism(value ?? throw new ArgumentException("Attestation cannot be null."));
+                if (value is X509Attestation)
+                {
+                    if ((((X509Attestation)value ?? throw new ArgumentException("Attestation cannot be null.")).ClientCertificates == null) &&
+                        (((X509Attestation)value).CAReferences == null))
+                    {
+                        throw new ArgumentException("Attestation mechanism do not contains client certificate or CA reference.");
```

Comment:
See above - this will be thrown for other nullable types of Attestation that could be null. (Which is the incorrect exception to throw in that case.)

Reply:
Done